app/_plugins/generators/seo/index_entry/non_product_page.rb

```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+module SEO
+  module IndexEntry
+    class NonProductPage < Base
+      def key
+        @key ||= @page.url
+      end
+
+      def process!(index)
+        super(index)
+
+        @page.data['canonical_url'] = @page.url
```

Comment:
Canonical URL should always have a trailing slash. Based on the link in the notification on https://deploy-preview-6733--kongdocs.netlify.app/mesh/dev/introduction/architecture/ we might need to enforce this here?

Reply:
hmmm, weird. both the meta tag and the link in the notification have a trailing slash, but it could be that  netlify is adding those, I'll take a look 

<img width="1240" alt="Screenshot 2024-01-12 at 09 19 35" src="https://github.com/Kong/docs.konghq.com/assets/715229/d0af9d19-4e78-4d3f-a789-f0c007be170a">